### PR TITLE
GET method for `ApolloSchemaDownloader`

### PR DESCRIPTION
**Pull request by mikepitre**
https://github.com/apollographql/apollo-ios/issues/1981

### CODE REVIEW

**Comment by calvincestari** [comment]
I've made some minor changes to the enum and added an error case for trying to use GET with the Apollo Registry. Tomorrow I'll add tests.

**Comment by calvincestari** [comment]
> Worth making this an option on the downloadMethod instead of on the initializer so you can't try to pass this in if you're using the registry?

Yes I'll take a pass at how that would look. There is an [error](https://github.com/apollographql/apollo-ios/pull/2010/commits/32dd79cb41aea5b4c0fceb0b796f06523478659e) for specifically this, but being able to completely rule out that possibility would be better.

**Comment by designatednerd** [comment]
Yeah error good, making it actually impossible better 😇

**Comment by calvincestari** [comment]
@designatednerd I think we're good to go with this PR now. There is more that I'd like to refactor with `ApolloSchemaDownloadConfiguration` but I'll do that in the 1.0 branch (#2012) because it will be breaking changes and right now supporting GET in this PR is all additive.

File: Sources/ApolloCodegenLib/ApolloSchemaDownloadConfiguration.swift [hunk 1]
```diff
@@ -33,11 +33,29 @@ public struct ApolloSchemaDownloadConfiguration {
         self.variant = variant
       }
     }
+
+    /// The HTTP request method.
```
Comment (designatednerd) [inline Sources/ApolloCodegenLib/ApolloSchemaDownloadConfiguration.swift:37]: probably worth clarifying that this is for introspection

File: Sources/ApolloCodegenLib/ApolloSchemaDownloadConfiguration.swift [hunk 2]
```diff
@@ -75,6 +93,7 @@ public struct ApolloSchemaDownloadConfiguration {
   /// - Parameters:
   ///   - downloadMethod: How to download your schema.
   ///   - downloadTimeout: The maximum time to wait before indicating that the download timed out, in seconds. Defaults to 30 seconds.
+  ///   - downloadHTTPMethod: HTTP Method for Introspection. Setting this to GET requires passing the name of the query parameter for the IntrospectionQuery. Defaults to POST.
```
Comment (designatednerd) [inline Sources/ApolloCodegenLib/ApolloSchemaDownloadConfiguration.swift:96]: I think this comment can also be removed

File: Sources/ApolloCodegenLib/ApolloSchemaDownloader.swift [hunk 3]
```diff
@@ -29,7 +31,11 @@ public struct ApolloSchemaDownloader {
       case .couldNotCreateSDLDataToWrite(let schema):
         return "Could not convert SDL schema into data to write to the filesystem. Schema: \(schema)"
       case .couldNotConvertIntrospectionJSONToSDL(let underlying):
-          return "Could not convert downloaded introspection JSON into SDL format. Underlying error: \(underlying)"
+        return "Could not convert downloaded introspection JSON into SDL format. Underlying error: \(underlying)"
+      case .couldNotCreateURLComponentsFromEndpointURL:
+        return "Could not create URLComponents from EndpointURL for Introspection."
+      case .couldNotGetURLFromURLComponents:
+        return "Could not get URL from URLComponents."
```
Comment (designatednerd) [inline Sources/ApolloCodegenLib/ApolloSchemaDownloader.swift:38]: Might be worth passing the components here so people can examine them. Also fine not to.

**Review by designatednerd — APPROVED** [review]
Couple suggestions, otherwise good to go!

**Pull request merged**